### PR TITLE
Add stubs for the App Check providers

### PR DESCRIPTION
**Pull request by a-maurice**
Add more App Check stubs for the default providers. They are split based on which platforms they should support.
***

Locally building the platforms
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [x] New feature. A non-breaking change which adds functionality.
- [ ] Other, such as a build process or documentation change.
***

#### Notes
- Bug fixes and feature changes require an update to the `Release Notes` section of `release_build_files/readme.md`.
- Read the contribution guidelines [CONTRIBUTING.md](https://github.com/firebase/firebase-cpp-sdk/blob/main/CONTRIBUTING.md).
- Changes to the public API require an internal API review. If you'd like to help us make Firebase APIs better, please propose your change in a feature request so that we can discuss it together.

### CODE REVIEW

**Review by AlmostMatt — APPROVED** [review]

File: app_check/src/android/debug_provider.cc [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    g_debug_app_check_provider_factory = new DebugAppCheckProviderFactory();
+  }
+  return g_debug_app_check_provider_factory;
+}
+
+DebugAppCheckProviderFactory::DebugAppCheckProviderFactory() {}
+
+DebugAppCheckProviderFactory::~DebugAppCheckProviderFactory() {}
+
+AppCheckProvider* DebugAppCheckProviderFactory::CreateProvider(App* app) {
```
[the remark this answers is not in the document]
Comment (a-maurice) [reply app_check/src/android/debug_provider.cc:34]: When they are properly implemented, yes, it should.  I didn't want to dive too deep into the actual implementation here.

File: app_check/src/android/play_integrity_provider.cc [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  }
+  return g_play_integrity_provider_factory;
+}
+
+PlayIntegrityProviderFactory::PlayIntegrityProviderFactory() {}
+
+PlayIntegrityProviderFactory::~PlayIntegrityProviderFactory() {}
+
+AppCheckProvider* PlayIntegrityProviderFactory::CreateProvider(App* app) {
+  return nullptr;
```
[the remark this answers is not in the document]
Comment (a-maurice) [reply app_check/src/android/play_integrity_provider.cc:35]: Same as the other one, leaving it null until the implementation.

**Review by AlmostMatt — COMMENTED** [review]
One relatively minor thing:
debug_provider.cc has the same file name for each platform.
I noticed that we often add _android _ios and _stub suffixes for such file names.

the other providers only have one non-stub implementation, but I suppose even for them the platform suffixes could still be done.

**Review by DellaBitta — APPROVED** [review]

**Comment by a-maurice** [comment]
> One relatively minor thing: debug_provider.cc has the same file name for each platform. I noticed that we often add _android _ios and _stub suffixes for such file names.
> 
> the other providers only have one non-stub implementation, but I suppose even for them the platform suffixes could still be done.

Yeah, I'm not quite sure why we do that, considering they are in folders, but I'll keep up the same pattern for consistency.

**Pull request merged**